phenol-analysis/src/main/java/org/monarchinitiative/phenol/analysis/mgsa/TermToItemMatrix.java

```diff
@@ -41,26 +38,26 @@
         n_annotated_terms = termToGeneMultimap.keySet().size();
         // First get list and indices of the Terms
         int i = 0;
-        ImmutableList.Builder<TermId> builder = new ImmutableList.Builder<>();
-        ImmutableMap.Builder<TermId,Integer> mapBuilder = new ImmutableMap.Builder<>();
+        List<TermId> builder = new ArrayList<>();
+        Map<TermId, Integer> mapBuilder = new HashMap<>();
         for (TermId ontologyTermId : termToGeneMultimap.keySet()) {
           builder.add(ontologyTermId);
           mapBuilder.put(ontologyTermId, i);
           i++;
         }
-        this.goTermList = builder.build();
-        this.goTermToIndexMap = mapBuilder.build();
+        this.goTermList = List.copyOf(builder);
+        this.goTermToIndexMap = Map.copyOf(mapBuilder);
         // Now get list and indices of the genes
-        builder = new ImmutableList.Builder<>();
-        mapBuilder = new ImmutableMap.Builder<>();
+        builder.clear();
+        mapBuilder.clear();
         i = 0;
         for (TermId gene : genes) {
           builder.add(gene);
           mapBuilder.put(gene, i);
           i++;
         }
-        this.annotatedItemList = builder.build();
-        this.annotatedItemToIndexMap = mapBuilder.build();
+        this.annotatedItemList = List.copyOf(builder);
+        this.annotatedItemToIndexMap = Map.copyOf(mapBuilder);
```

Comment:
@ielis I know you're just upgrading the Java 11 usage, but this is unnecessarily complicated. e.g. lines 51-59 could really just be replaced with a simple call to `List.copyOf()`

```java
this.annotatedItemList = List.copyOf(genes);
```

meaning there is no re-use of the now oddly-named `builder` variable.

Reply:
Yes, this should work. Thanks!